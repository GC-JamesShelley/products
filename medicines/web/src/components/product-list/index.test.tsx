import { shallow } from 'enzyme';
import React from 'react';
import ProductList from './index';

describe(ProductList, () => {
  it('should render', () => {
    const component = shallow(
      <ProductList
        title={'Coffee'}
        products={[
          { name: 'Caffe Latte', count: 0 },
          { name: 'Americano', count: 1 },
          { name: 'Flat White', count: -1 },
          { name: 'Mocha', count: 9999999 },
          { name: 'Cafe Creme', count: -9999999 },
        ]}
      />,
    );
    expect(component).toMatchSnapshot();
  });
});
