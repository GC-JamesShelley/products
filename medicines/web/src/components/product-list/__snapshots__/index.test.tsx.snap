// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrugIndex should render 1`] = `
<styled.nav>
  <p
    className="horizontal"
  >
    Coffee
  </p>
  <ul
    className="horizontal"
  >
    <li
      className=""
      key="A"
    >
      <Link
        href="/substance-index?letter=A"
      >
        <a>
          A
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="B"
    >
      <Link
        href="/substance-index?letter=B"
      >
        <a>
          B
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="C"
    >
      <Link
        href="/substance-index?letter=C"
      >
        <a>
          C
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="D"
    >
      <Link
        href="/substance-index?letter=D"
      >
        <a>
          D
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="E"
    >
      <Link
        href="/substance-index?letter=E"
      >
        <a>
          E
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="F"
    >
      <Link
        href="/substance-index?letter=F"
      >
        <a>
          F
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="G"
    >
      <Link
        href="/substance-index?letter=G"
      >
        <a>
          G
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="H"
    >
      <Link
        href="/substance-index?letter=H"
      >
        <a>
          H
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="I"
    >
      <Link
        href="/substance-index?letter=I"
      >
        <a>
          I
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="J"
    >
      <Link
        href="/substance-index?letter=J"
      >
        <a>
          J
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="K"
    >
      <Link
        href="/substance-index?letter=K"
      >
        <a>
          K
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="L"
    >
      <Link
        href="/substance-index?letter=L"
      >
        <a>
          L
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="M"
    >
      <Link
        href="/substance-index?letter=M"
      >
        <a>
          M
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="N"
    >
      <Link
        href="/substance-index?letter=N"
      >
        <a>
          N
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="O"
    >
      <Link
        href="/substance-index?letter=O"
      >
        <a>
          O
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="P"
    >
      <Link
        href="/substance-index?letter=P"
      >
        <a>
          P
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="Q"
    >
      <Link
        href="/substance-index?letter=Q"
      >
        <a>
          Q
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="R"
    >
      <Link
        href="/substance-index?letter=R"
      >
        <a>
          R
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="S"
    >
      <Link
        href="/substance-index?letter=S"
      >
        <a>
          S
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="T"
    >
      <Link
        href="/substance-index?letter=T"
      >
        <a>
          T
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="U"
    >
      <Link
        href="/substance-index?letter=U"
      >
        <a>
          U
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="V"
    >
      <Link
        href="/substance-index?letter=V"
      >
        <a>
          V
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="W"
    >
      <Link
        href="/substance-index?letter=W"
      >
        <a>
          W
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="X"
    >
      <Link
        href="/substance-index?letter=X"
      >
        <a>
          X
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="Y"
    >
      <Link
        href="/substance-index?letter=Y"
      >
        <a>
          Y
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="Z"
    >
      <Link
        href="/substance-index?letter=Z"
      >
        <a>
          Z
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="0"
    >
      <Link
        href="/substance-index?letter=0"
      >
        <a>
          0
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="1"
    >
      <Link
        href="/substance-index?letter=1"
      >
        <a>
          1
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="2"
    >
      <Link
        href="/substance-index?letter=2"
      >
        <a>
          2
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="3"
    >
      <Link
        href="/substance-index?letter=3"
      >
        <a>
          3
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="4"
    >
      <Link
        href="/substance-index?letter=4"
      >
        <a>
          4
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="5"
    >
      <Link
        href="/substance-index?letter=5"
      >
        <a>
          5
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="6"
    >
      <Link
        href="/substance-index?letter=6"
      >
        <a>
          6
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="7"
    >
      <Link
        href="/substance-index?letter=7"
      >
        <a>
          7
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="8"
    >
      <Link
        href="/substance-index?letter=8"
      >
        <a>
          8
           
        </a>
      </Link>
    </li>
    <li
      className=""
      key="9"
    >
      <Link
        href="/substance-index?letter=9"
      >
        <a>
          9
           
        </a>
      </Link>
    </li>
  </ul>
</styled.nav>
`;

exports[`ProductList should render 1`] = `
<styled.nav>
  <h3>
    Coffee
  </h3>
  <ul>
    <li
      className="product-name"
      key="Caffe Latte"
    >
      <Link
        href="/product?product=Caffe%20Latte"
      >
        <a>
          Caffe Latte
           
          0
        </a>
      </Link>
    </li>
    <li
      className="product-name"
      key="Americano"
    >
      <Link
        href="/product?product=Americano"
      >
        <a>
          Americano
           
          (
          1
           files)
        </a>
      </Link>
    </li>
    <li
      className="product-name"
      key="Flat White"
    >
      <Link
        href="/product?product=Flat%20White"
      >
        <a>
          Flat White
           
          (
          -1
           files)
        </a>
      </Link>
    </li>
    <li
      className="product-name"
      key="Mocha"
    >
      <Link
        href="/product?product=Mocha"
      >
        <a>
          Mocha
           
          (
          9999999
           files)
        </a>
      </Link>
    </li>
    <li
      className="product-name"
      key="Cafe Creme"
    >
      <Link
        href="/product?product=Cafe%20Creme"
      >
        <a>
          Cafe Creme
           
          (
          -9999999
           files)
        </a>
      </Link>
    </li>
  </ul>
</styled.nav>
`;
